ruma-events-macros/src/event_enum.rs

```diff
@@ -180,10 +188,26 @@ fn to_event_path(name: &LitStr) -> TokenStream {
         .map(|s| s.chars().next().unwrap().to_uppercase().to_string() + &s[1..])
         .collect::<String>();
 
-    let content_str = Ident::new(&format!("{}Event", event), span);
     let path = path.iter().map(|s| Ident::new(s, span));
-    quote! {
-        ::ruma_events::#( #path )::*::#content_str
+
+    match struct_name.to_string().as_str() {
+        "MessageEvent" | "MessageEventStub" if event_str.contains("redaction") => {
```

Comment:
```suggestion
        "MessageEvent" | "MessageEventStub" if event_str.contains("redaction") => {
```

Why use `event_str.contains` rather than match the full redaction event type?

Reply:
Meaning `event_str == "m.room.readaction" ? that makes more sense I guess being leaniant makes no sense here.